### DISCUSSION TITLE
Asset type conversions does not work for things that extent Image or Bitmap

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [AvaloniaUI/Avalonia#16496] [Q&A]
### Describe the bug

I was trying to write a IndexedBitmap class that I use in xaml, however it seems Avaloina does not take in account classes derived from Bitmap and Image when converting a path.

### To Reproduce

- Make a simple bitmapB class that extends from Bitmap.
- Use it as an observable type in your view model.
- Try to bind a image to it via file path in xaml.
- See there is not auto complete or change.

### Expected behavior

Something like this should happen.
![image](https://github.com/user-attachments/assets/1099833b-aeed-4663-b5da-b55f0a9df2f9)

### Avalonia version

11.0.10

### OS

Linux

### Additional context

I am using this to convert to writable bitmap from Bitmap.
```cs
    public static WriteableBitmap ToWriteable(this Bitmap bitmap)
    {
        var writeable = new WriteableBitmap(bitmap.PixelSize, bitmap.Dpi, bitmap.Format, bitmap.AlphaFormat);
        using var buffer    = writeable.Lock();
        bitmap.CopyPixels(buffer, AlphaFormat.Premul);
        return writeable;
    }
```

**Comment by maxkatz6** [comment]
How would Avalonia know how to create instance of bitmapB from string?

For custom controls and types, you task typically is solved by creating a TypeConverter that defines how to parse string and create a `bitmapB` from it.
And this type converter then is used as a attribute on `bitmapB` type. See https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.typeconverter?view=net-8.0 for more details.

XAML compiler will find this type converter and use it to create a bitmap from string.,

**Reply by Shadowblitz16** [reply]
Is there a way to inherit the file parsing behavior?
I really don't want to manually parse file paths in rider.
